wgpu/tests/common/mod.rs

```diff
@@ -314,6 +330,60 @@ pub fn initialize_test(parameters: TestParameters, test_function: impl FnOnce(Te
     }
 }
 
+#[cfg(not(all(target_arch = "wasm32", feature = "webgl")))]
+fn initialize_adapter() -> Adapter {
+    let backend_bits = wgpu::util::backend_bits_from_env().unwrap_or_else(Backends::all);
+    let instance = Instance::new(backend_bits);
+
+    pollster::block_on(wgpu::util::initialize_adapter_from_env_or_default(
+        &instance,
+        backend_bits,
+        None,
+    ))
+    .expect("could not find suitable adapter on the system")
+}
+
+#[cfg(all(target_arch = "wasm32", feature = "webgl"))]
+fn initialize_adapter() -> Adapter {
```

Comment:
We should be able to unify these two functions by having a few conditional lines of code. initialize_adapter_from_env_or_default and friends should work on web, will just always fallback to the default.